ui/src/alerting/components/builder/RemoveButton.tsx

```diff
@@ -0,0 +1,53 @@
+// Libraries
+import React, {FunctionComponent} from 'react'
+import {connect} from 'react-redux'
+
+// Components
+import {Button} from '@influxdata/clockface'
+
+// Actions
+import {setType as setViewType} from 'src/timeMachine/actions'
+import {setCurrentCheck} from 'src/alerting/actions/checks'
+import {setActiveTab} from 'src/timeMachine/actions'
+
+//Types
+import {RemoteDataState} from 'src/types'
+
+interface DispatchProps {
+  setActiveTab: typeof setActiveTab
+  setViewType: typeof setViewType
+  setCurrentCheck: typeof setCurrentCheck
+}
+
+type Props = DispatchProps
+
+const RemoveButton: FunctionComponent<Props> = ({
+  setActiveTab,
+  setViewType,
+  setCurrentCheck,
+}) => {
+  const handleClick = () => {
+    setCurrentCheck(RemoteDataState.NotStarted, null)
```

Comment:
I think one of the main benefits of redux is that it decouples emitting actions from the state transformations. But when we write actions, we often treat them as simple setter functions for some single field in a redux slice. By doing so, I think we are getting all of the indirection of the event emitter abstraction with none of the benefits. 

In this case, a single action was taken in the UI (the alert builder remove button was clicked), but we have to emit 3 different actions. Concretely, this causes an issue where every Redux connected component will render 3 times instead of one, and the intermediate renders will see partially set state. It also means that the state transitions in our UI are spread all over the codebase, rather than being centralized in a reducer (so that we are coding an implicit state machine, rather than an explicit one).

This feels like a really widespread issue in our codebase... but I'd rather not continue it if possible. Could we instead make a single action (`ALERT_BUILDER_REMOVE_BUTTON_CLICKED` or whatever), and put this logic in a reducer?